PurchasesTests/Caching/InMemoryCachedObjectTests.swift

```diff
@@ -0,0 +1,124 @@
+//
+// Created by RevenueCat on 2/4/20.
+// Copyright (c) 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+import Nimble
+import XCTest
+
+@testable import Purchases
+
+class InMemoryCachedObjectTests: XCTestCase {
+
+    // mark: isCacheStale
+
+    func testIsCacheStaleIsFalseBeforeDurationExpires() {
+        let cacheDurationInSeconds: Int32 = 5
+
+        let cachedObject = RCInMemoryCachedObject<NSString>(cacheDurationInSeconds: cacheDurationInSeconds,
+                                                            lastUpdatedAt: nil)
+        cachedObject.stubbedNow = Date()
+
+        cachedObject.cacheInstance("myString", date: Date())
+
+        expect(cachedObject.isCacheStale()) == false
+
+        cachedObject.stubbedNow = Date(timeIntervalSinceNow: 4)
+
+        expect(cachedObject.isCacheStale()) == false
+    }
+
+    func testIsCacheStaleIsTrueAfterDurationExpires() {
+        let cacheDurationInSeconds: Int32 = 5
+
+        let cachedObject = RCInMemoryCachedObject<NSString>(cacheDurationInSeconds: cacheDurationInSeconds,
+                                                            lastUpdatedAt: nil)
+        cachedObject.stubbedNow = Date()
+
+        cachedObject.cacheInstance("myString", date: Date())
+
+        expect(cachedObject.isCacheStale()) == false
+
+        cachedObject.stubbedNow = Date(timeIntervalSinceNow: 6)
+
+        expect(cachedObject.isCacheStale()) == true
+    }
+
+    // mark: clearCacheTimestamp
+
+    func testClearCacheTimestampClearsCorrectly() {
+        let cacheDurationInSeconds: Int32 = 5
+
+        let cachedObject = RCInMemoryCachedObject<NSString>(cacheDurationInSeconds: cacheDurationInSeconds,
+                                                            lastUpdatedAt: nil)
+        cachedObject.cacheInstance("myString", date: Date())
+        expect(cachedObject.lastUpdatedAt).toNot(beNil())
+        expect(cachedObject.cachedInstance()).toNot(beNil())
+
+        cachedObject.clearCacheTimestamp()
+        expect(cachedObject.lastUpdatedAt).to(beNil())
+        expect(cachedObject.cachedInstance()).toNot(beNil())
+    }
+
+    // mark: clearCache
+
+    func testClearCacheClearsCorrectly() {
+        let cacheDurationInSeconds: Int32 = 5
+
+        let cachedObject = RCInMemoryCachedObject<NSString>(cacheDurationInSeconds: cacheDurationInSeconds,
+                                                            lastUpdatedAt: nil)
+        cachedObject.cacheInstance("myString", date: Date())
```

Comment:
The parameter `date` in the `cacheInstance` function is confusing, I would rename to something more descriptive. Also, I would not over-engineer it and if it's always being set to `Date()` I would just set it inside the method to `Date()`.

Reply:
agreed, killed